### PR TITLE
[WIP] QuantizedConvolution Builder

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: src/ngraph/runtime/cpu/builder/quantized_conv.cpp [hunk 1]
```diff
@@ -37,6 +37,7 @@ namespace ngraph
             {
                 if (runtime::cpu::mkldnn_utils::use_mkldnn_kernel(node))
                 {
+                    auto qconv = static_cast<const ngraph::op::QuantizedConvolution*>(node);
```
Comment (arogowie-intel) [inline src/ngraph/runtime/cpu/builder/quantized_conv.cpp]: The compiler fails at this and below lines: https://aipg-rancher.intel.com/jenkins/algo/blue/organizations/jenkins/ngraph-ci-premerge/detail/PR-2351/1/pipeline/74#step-523-log-160

File: src/ngraph/builder/quantization/quantized_linear_convolution.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
```
Comment (arogowie-intel) [inline src/ngraph/builder/quantization/quantized_linear_convolution.cpp]: Update year.

File: src/ngraph/builder/quantization/quantized_linear_convolution.cpp [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                                           const Strides& data_dilation_strides,
+                                           std::shared_ptr<Node> input_scale,
+                                           std::shared_ptr<Node> input_zero_point,
+                                           std::shared_ptr<Node> filter_scale,
+                                           std::shared_ptr<Node> filter_zero_point,
+                                           std::shared_ptr<Node> output_scale,
+                                           std::shared_ptr<Node> output_zero_point)
+            {
+                // TODO: need to handle the case where offset is provided (assuming 0)
+                // TODO: need to establish cross-nGraph view of scale (mult or div)
```
Comment (arogowie-intel) [inline src/ngraph/builder/quantization/quantized_linear_convolution.cpp]: What about these comments?

File: src/ngraph/builder/quantization/quantized_linear_convolution.cpp [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                                               const Strides& data_dilation_strides,
+                                               std::shared_ptr<Node> input_scale,
+                                               std::shared_ptr<Node> input_zero_point,
+                                               std::shared_ptr<Node> filter_scale,
+                                               std::shared_ptr<Node> filter_zero_point,
+                                               std::shared_ptr<Node> output_scale,
+                                               std::shared_ptr<Node> output_zero_point)
+            {
+                // TODO: need to handle the case where offset is provided (assuming 0)
+                // TODO: need to establish cross-nGraph view of scale (mult or div)
```
Comment (arogowie-intel) [inline src/ngraph/builder/quantization/quantized_linear_convolution.cpp]: What about these comments?

File: src/ngraph/builder/quantization/quantized_linear_convolution.hpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
```
Comment (arogowie-intel) [inline src/ngraph/builder/quantization/quantized_linear_convolution.hpp]: Update year.

File: src/ngraph/builder/quantization/quantized_linear_convolution.cpp (hunk 3 above)
Comment (tsocha) [reply src/ngraph/builder/quantization/quantized_linear_convolution.cpp]: @adstraw

File: src/ngraph/builder/quantization/quantized_linear_convolution.cpp (hunk 4 above)
Comment (tsocha) [reply src/ngraph/builder/quantization/quantized_linear_convolution.cpp]: @adstraw

**Comment by mbrookhart** [comment]
Can you add a description? I'm not sure how this relates to the quantized builders we already have and https://github.com/NervanaSystems/ngraph/pull/2292

File: src/ngraph/runtime/cpu/pass/cpu_assignment.cpp [hunk 6]
```diff
@@ -660,11 +660,7 @@ namespace ngraph
                 template <>
                 void CPUAssignment::ASSIGN_DECL(ngraph::op::QuantizedConvolution)
                 {
-                    if (node->get_input_element_type(0) == element::u8 &&
-                        node->get_input_element_type(1) == element::i8)
-                    {
-                        runtime::cpu::mkldnn_utils::assign_mkldnn_kernel(node);
-                    }
+                    runtime::cpu::mkldnn_utils::assign_mkldnn_kernel(node);
```
Comment (nbpatel) [inline src/ngraph/runtime/cpu/pass/cpu_assignment.cpp]: check if ((input == u8 || input == s8) && filter == s8)

File: src/ngraph/runtime/cpu/pass/cpu_assignment.cpp [hunk 7]
```diff
@@ -680,11 +676,7 @@ namespace ngraph
                 template <>
                 void CPUAssignment::ASSIGN_DECL(ngraph::op::QuantizedConvolutionBias)
                 {
-                    if (node->get_input_element_type(0) == element::u8 &&
-                        node->get_input_element_type(1) == element::i8)
-                    {
-                        runtime::cpu::mkldnn_utils::assign_mkldnn_kernel(node);
-                    }
+                    runtime::cpu::mkldnn_utils::assign_mkldnn_kernel(node);
```
Comment (nbpatel) [inline src/ngraph/runtime/cpu/pass/cpu_assignment.cpp]: same as above

File: src/ngraph/builder/quantization/quantized_linear_convolution.hpp [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+#include "ngraph/coordinate_diff.hpp"
+#include "ngraph/node.hpp"
+
+namespace ngraph
+{
+    namespace builder
+    {
+        namespace quantization
+        {
+            std::shared_ptr<Node>
```
Comment (mbrookhart) [inline src/ngraph/builder/quantization/quantized_linear_convolution.hpp]: Documentaiton?

File: src/ngraph/builder/quantization/quantized_linear_convolution.hpp [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                                           const CoordinateDiff& padding_above,
+                                           const Strides& data_dilation_strides,
+                                           std::shared_ptr<Node> input_scale,
+                                           std::shared_ptr<Node> input_zero_point,
+                                           std::shared_ptr<Node> filter_scale,
+                                           std::shared_ptr<Node> filter_zero_point,
+                                           std::shared_ptr<Node> output_scale,
+                                           std::shared_ptr<Node> output_zero_point);
+
+            std::shared_ptr<Node>
```
Comment (mbrookhart) [inline src/ngraph/builder/quantization/quantized_linear_convolution.hpp]: Documentaiton?

Comment (postrational) [reply src/ngraph/builder/quantization/quantized_linear_convolution.hpp]: +1

**Pull request closed without merging**